File: mobile/src/main/java/com/cos/mobile/model/Order.java
package com.cos.mobile.model;

import java.sql.Date;
import java.sql.Timestamp;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.hibernate.annotations.CreationTimestamp;
import org.springframework.format.annotation.DateTimeFormat;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;


@Table(name="Orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Order {
	@Id // 기본키
	private String id;
	
	private String color;
	
	private String storage;
	
	private String beforetel;
	
	private String aftertel;
	
	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name="telfeeid")
	private TelecomFee telfee;
	
	@ManyToOne
	@JoinColumn(name="productid")
	private Product product;	// 상품id로 join
	
	@ManyToOne(fetch = FetchType.EAGER, optional = true)
	@JoinColumn(name="userid")
	private Users user;
	
	@DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss.S")
	private Date birth;
	
	private String newphone;
	
	private String posttel;
	
	private String postname;
	
	private String email;
	
	private String dcchoice;
	
	private int dc_option;
	
	@CreationTimestamp
	private Timestamp createDate;
	
	private String status;
	
	private String zipcode;
	
	private String address;
	
	private String addrdetail;
	
	private String requestmsg;	// 요청사항
}

File: mobile/src/main/resources/static/script/order.js
$(document).on("click", "#btn-updateOrder", function(){
	var id=$("#orderid").val();
	console.log(id);
	let data={
			status: $("#updateDelivery").val()
		};
		console.log(data);
		$.ajax({ 
			type:"PUT",
			url:"/api/updateDelivery/"+id,
			data:JSON.stringify(data),
			contentType:"application/json; charset=utf-8",
			dataType:"json",
		}).done(function(resp){
			alert("주문 상태 변경 완료!");
			location.href="/admin/delivery";
		}).fail(function(error){
			alert(JSON.stringify(error));
		});
});




$(document).on("click", "#btn_order", function(){
	let data={
			cartid: $("#cartid").val(),
			color: $("#color").val(),
			storage: $("#storage").val(),
			beforetel: $("#beforetel").val(),
			aftertel: $("#aftertel").val(),
			telfeeid: $("#telfeeid").val(),
			productid: $("#productid").val(),
			userid: $("#userid").val(),
			dcchoice: $("#dcchoice").val(),
			dc_option: $("#dc_option").val(),
			period: $("#period").val(),
			birth: $("#birth").val(),
			newphone: $("#newphone").val(),
			email: $("#email").val(),
			postName: $("#postName").val(),
			postTel: $("#postTel").val(),
			zipcode: $("#zipcode").val(),
			address: $("#address").val(),
			addrdetail: $("#addrdetail").val(),
			requestmsg: $("#requestmsg").val()
		};
		console.log(data);
		$.ajax({ 
			type:"POST",
			url:"/api/order",
			data:JSON.stringify(data),
			contentType:"application/json; charset=utf-8",
			dataType:"json",
			traditional:true
		}).done(function(resp){
			alert("주문이 완료되었습니다.");
			location.href="/delivery";
		}).fail(function(error){
			alert(JSON.stringify(error));
		});
});